lldb-eval/parser.cc

```diff
@@ -1323,4 +1616,698 @@ ExprResult Parser::ParseIntegerLiteral(clang::NumericLiteralParser& literal,
   return std::make_unique<LiteralNode>(std::move(value));
 }
 
+ExprResult Parser::BuildCStyleCast(Type type, ExprResult rhs,
+                                   clang::SourceLocation location) {
+  CStyleCastKind kind;
+  Type rhs_type = rhs->result_type_deref();
+
+  // Cast to basic type (integer/float).
+  if (type.IsScalar()) {
+    // Pointers can be cast to integers of the same or larger size.
+    if (rhs_type.IsPointerType() || rhs_type.IsNullPtrType()) {
+      // C-style cast from pointer to float/double is not allowed.
+      if (type.IsFloat()) {
+        BailOut(ErrorCode::kInvalidOperandType,
+                llvm::formatv("C-style cast from '{0}' to '{1}' is not allowed",
+                              rhs_type.GetName(), type.GetName()),
+                location);
+        return std::make_unique<ErrorNode>();
+      }
+      // Check if the result type is at least as big as the pointer size.
+      // TODO(werat): Use target pointer size, not host.
+      if (type.GetByteSize() < sizeof(void*)) {
+        BailOut(ErrorCode::kInvalidOperandType,
+                llvm::formatv(
+                    "cast from pointer to smaller type '{0}' loses information",
+                    type.GetName()),
+                location);
+        return std::make_unique<ErrorNode>();
+      }
+    }
+    // Otherwise accept only arithmetic types and enums.
```

Comment:
Nit: Perhaps put the comment inside the then block. Otherwise, it looks like the previous 'if' was already finished. (Here and below.)